modules/oe_subscriptions_digest/oe_subscriptions_digest.module

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @file
+ * Subscriptions digest module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\oe_subscriptions_digest\SubscriptionsFormAlter;
+
+/**
+ * Implements hook_flag_insert().
+ *
+ * The message_subscribe_email module uses message_subscribe to email
+ * subscribers that want updates. Every flag that is prefixed with "subscribe_"
+ * (a subscription flag) requires a flag that is prefixed with "email_" and the
+ * same suffix (e.g. "subscribe_node" and "email_node"). The flag should have at
+ * least one bundle associated with it.
+ */
+function oe_subscriptions_digest_flag_insert(EntityInterface $entity) {
+  $mail_prefix = \Drupal::config('message_subscribe_email.settings')->get('flag_prefix');
+  if (empty($mail_prefix) || !str_starts_with($entity->id(), 'subscribe_')) {
+    return;
+  }
+  $email_flag_id = str_replace('subscribe_', $mail_prefix . '_', $entity->id());
+  if (Drupal::service('entity_type.manager')->getStorage('flag')->load($email_flag_id)) {
+    return;
+  }
+  $entity->createDuplicate()
+    ->set('id', $email_flag_id)
+    ->set('label', $mail_prefix . ' - ' . $entity->label())
+    ->save();
+}
+
+/**
+ * Implements hook_flag_delete().
+ */
+function oe_subscriptions_digest_flag_delete(EntityInterface $entity) {
+  $mail_prefix = \Drupal::config('message_subscribe_email.settings')->get('flag_prefix');
+  if (empty($mail_prefix) || !str_starts_with($entity->id(), 'subscribe_')) {
+    return;
+  }
+  $email_flag_id = str_replace('subscribe_', $mail_prefix . '_', $entity->id());
+  if ($email_flag = \Drupal::service('entity_type.manager')->getStorage('flag')->load($email_flag_id)) {
+    $email_flag->delete();
+  }
+}
+
+/**
+ * Implements hook_user_update().
+ */
+function oe_subscriptions_digest_user_update(EntityInterface $entity) {
+  if (!$entity->hasField('message_digest') ||
+    $entity->get('message_digest')->value === $entity->original->get('message_digest')->value) {
+    return;
+  }
+  $flagging_storage = \Drupal::service('entity_type.manager')->getStorage('flagging');
+  $query = $flagging_storage->getQuery();
+  $query
+    ->accessCheck(FALSE)
+    ->condition('flag_id', 'subscribe_', 'STARTS_WITH')
```

Comment:
The message digest is stored on email flags, not the subscribe one.

Reply:
Changed